app/src/main/java/ci/CIServer.java

```diff
@@ -16,8 +25,150 @@
 		GITHUB_TOKEN = System.getenv("GITHUB_TOKEN");
 	}
 
+
+	/**
+	 * Handles a push event
+	 * */
 	public void handle(HttpExchange exchange) throws IOException {
+
+		JSONArray commits = getCommits(exchange);
+		ArrayList<Boolean> isBuildSuccessful = buildCommits(commits);
+
 		exchange.sendResponseHeaders(200, 0);
 		return;
 	}
+
+	/**
+	 * Returns a JSONArray object of commits from the JSON body of a POST request
+	 * See Github push event API for detailed info
+	 * */
+	private static JSONArray getCommits(HttpExchange exchange) {
+	
+		try {
+			//format http json body
+			InputStream iStream = exchange.getRequestBody();
+			InputStreamReader iStreamReader = new InputStreamReader(iStream);
+			JSONParser jsonParser = new JSONParser();
+			JSONObject jsonObject = (JSONObject)jsonParser.parse(iStreamReader);
+
+			//return the commits as a JSONArray object
+			JSONArray commits = (JSONArray)jsonObject.get("commits");
+			return commits;
+		}
+		catch(ParseException e) {
+			System.out.println("Failed to parse JSON body.");
+			System.exit(0);
+		}
+		catch(IOException e) {
+			System.out.println("Failed to parse InputStreamReader.");
+			System.exit(0);
+		}
+
+		return null;
+	}
+
+	/**
+	 * Returns an ArrayList of booleans representing whether a commit successfully
+	 * compiles or not.
+	 * */
+	private static ArrayList<Boolean> buildCommits(JSONArray commits) {
+
+		try {
+			//temporarily create builds directory
+			Runtime.getRuntime().exec("mkdir builds");
+		}
+		catch(IOException e) {
+			System.out.println("Failed to create new builds directory.");
+			System.exit(0);
+		}
+
+		ArrayList<Boolean> didBuild = new ArrayList<Boolean>();
+
+		for(int i = 0; i < commits.size(); i++) {
+			JSONObject commit = (JSONObject)commits.get(i);
+			String sha = (String)commit.get("id");			//sha of commit
+
+			didBuild.add(buildCommit(sha));
+		}
+
+		try {
+			//clean up created builds directory
+			Runtime.getRuntime().exec("rm -rf builds");
+		}
+		catch(IOException e) {
+			System.out.println("Failed to clean up created builds directory.");
+			System.exit(0);
+		}
+
+		return didBuild;
+	}
+
+	/**
+	 * Returns a boolean value representing whether a specific commit compiles successfully.
+	 * sha parameter is used to locally build the project using the gradle wrapper.
+	 * The output is parsed to determine whether build was successful or not.
+	 * */
+	private static boolean buildCommit(String sha) {
+
+		try {
+			//create folder for commit
+			Runtime.getRuntime().exec("mkdir builds/"+sha);
+		}
+		catch(IOException e) {
+			System.out.println("Failed to create folder for commit.");
+			System.exit(0);
+		}
+
+		try {
+			//clone repo
+			String repo = "https://github.com/DanielH4/Assignment-2-CI.git";
```

Comment:
May be it should be https://github.com/DD2480-Group-11/Assignment-2-CI.git in the main repo.